coreclient/src/clients/message.rs

```diff
@@ -0,0 +1,186 @@
+// SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use anyhow::Context;
+use phnxtypes::{
+    identifiers::QualifiedUserName, messages::client_ds_out::SendMessageParamsOut, time::TimeStamp,
+};
+use rusqlite::{Connection, Transaction};
+
+use crate::{Conversation, ConversationId, ConversationMessage, MimiContent};
+
+use super::{ApiClients, CoreUser, Group, StoreNotifier};
+
+impl CoreUser {
+    /// Send a message and return it.
+    ///
+    /// Note that the message has already been sent to the DS and has internally been stored in the
+    /// conversation store.
+    pub(crate) async fn send_message(
+        &self,
+        conversation_id: ConversationId,
+        content: MimiContent,
+    ) -> anyhow::Result<ConversationMessage> {
+        let unsent_group_message = self
+            .with_transaction(|transaction| {
+                InitialParams {
+                    conversation_id,
+                    content,
+                }
+                .store_unsent_message(transaction, self.store_notifier(), &self.user_name())?
+                .create_unsent_group_message(transaction, self.store_notifier())
```

Comment:
Technically, this also contains access to the crypto provider. Should we pull it up? @kkohbrok

Reply:
I think that would be good, yes.